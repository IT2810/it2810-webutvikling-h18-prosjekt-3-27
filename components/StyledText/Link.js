import React, {Component} from "react";
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity
} from "react-native";

export default class Link extends Component {

  getStylesArrayFromProps() {
    return this.props.style || [];
  }

  render() {
    return (
      <Text
        onPress={this.props.onPress}
        style={[...this.getStylesArrayFromProps(), styles.linkText]}>
        {this.props.children}
      </Text>
    );
  }
}

const styles = StyleSheet.create({
  linkText: {
    color: '#2e78b7',
    textDecorationLine: "underline"
  }
});
