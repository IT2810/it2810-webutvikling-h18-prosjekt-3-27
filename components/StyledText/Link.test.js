import React from "react";
import renderer from "react-test-renderer";
import Link from "./Link";

it("renders correctly with no props", () => {
  const tree = renderer
    .create(<Link/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly with content", () => {
  const tree = renderer
    .create(<Link>This is some link text</Link>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly with content and handler", () => {
  const tree = renderer
    .create(<Link onPress={() => console.log("Example handler")}>This is some link text</Link>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly with handler", () => {
  const tree = renderer
    .create(<Link onPress={() => {console.log("Some other handler")}}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


it("renders correctly and overwrites color", () => {
  const tree = renderer
    .create(<Link style={{color: "#fff"}}>This should be blue</Link>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
