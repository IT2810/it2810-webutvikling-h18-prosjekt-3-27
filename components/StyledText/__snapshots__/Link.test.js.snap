// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and overwrites color 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#2e78b7",
        "textDecorationLine": "underline",
      },
    ]
  }
>
  This should be blue
</Text>
`;

exports[`renders correctly with content 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#2e78b7",
        "textDecorationLine": "underline",
      },
    ]
  }
>
  This is some link text
</Text>
`;

exports[`renders correctly with content and handler 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  onPress={[Function]}
  style={
    Array [
      Object {
        "color": "#2e78b7",
        "textDecorationLine": "underline",
      },
    ]
  }
>
  This is some link text
</Text>
`;

exports[`renders correctly with handler 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  onPress={[Function]}
  style={
    Array [
      Object {
        "color": "#2e78b7",
        "textDecorationLine": "underline",
      },
    ]
  }
/>
`;

exports[`renders correctly with no props 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#2e78b7",
        "textDecorationLine": "underline",
      },
    ]
  }
/>
`;
